### PR TITLE
Adding HoleM63

### PR DESCRIPTION
**Pull request by clementd08**
Hello all,

This Pull Request introduce a new Hole,
HoleM63

Top_flat : 
HoleM63_mag_int_rotor_top_flat
Normal :
HoleM63_mag_int_rotor
This Hole has the particularity to have two schematics with possibility to change beetween normal and top flat.

Best regards,
Clément

### CODE REVIEW

File: Tests/Methods/Slot/test_HoleM62_meth.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+from pyleecan.Classes.Magnet import Magnet
+from numpy import exp, arcsin, ndarray, pi
+import matplotlib.pyplot as plt
+
+# For AlmostEqual
+DELTA = 1e-4
+
+
+HoleM62_test = list()
```
Comment (AdrienLeduqueEomys) [inline Tests/Methods/Slot/test_HoleM62_meth.py:19]: Majuscule

File: Tests/GUI/Dialog/DMachineSetup/PHole/test_PHoleM63.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    @classmethod
+    def teardown_class(cls):
+        """Exit the app after the test"""
+        cls.app.quit()
+
+    def setup_method(self):
+        """Run at the begining of every test to setup the gui"""
+        test_obj = LamHole(Rint=0.1, Rext=0.5)
+        test_obj.hole = list()
```
Comment (AdrienLeduqueEomys) [inline Tests/GUI/Dialog/DMachineSetup/PHole/test_PHoleM63.py:34]: Pourquoi faire une list ?

File: Tests/Methods/Slot/test_HoleM63_meth.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+from pyleecan.Classes.Magnet import Magnet
+from numpy import exp, arcsin, ndarray, pi
+import matplotlib.pyplot as plt
+
+# For AlmostEqual
+DELTA = 1e-4
+
+
+HoleM63_test = list()
```
Comment (AdrienLeduqueEomys) [inline Tests/Methods/Slot/test_HoleM63_meth.py:19]: majuscule

File: Tests/Methods/Slot/test_HoleM63_meth.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+    @pytest.mark.parametrize("test_dict", HoleM63_test)
+    def test_comp_surface(self, test_dict):
+        """Check that the computation of the surface is correct"""
+        test_obj = test_dict["test_obj"]
+        result = Hole.comp_surface(test_obj.hole[0])
+
+        a = result
+        b = test_dict["S_exp"]
+        msg = "Return " + str(a) + " expected " + str(b)
```
Comment (AdrienLeduqueEomys) [inline Tests/Methods/Slot/test_HoleM63_meth.py:101]: f-string

File: pyleecan/GUI/Dialog/DMachineSetup/SMHoleMag/PHoleM63/PHoleM63.py [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+        if self.check() is None:
+            try:
+                # We compute the output only if the hole is correctly set
+                # Compute all the needed output as string
+                s_surf = format(self.u.get_m2(self.hole.comp_surface()), ".4g")
+                m_surf = format(self.u.get_m2(self.hole.comp_surface_magnets()), ".4g")
+
+                # Update the GUI to display the Output
+                self.out_slot_surface.setText(
+                    "Slot suface (2 part): " + s_surf + " " + self.u.get_m2_name()
```
Comment (AdrienLeduqueEomys) [inline pyleecan/GUI/Dialog/DMachineSetup/SMHoleMag/PHoleM63/PHoleM63.py:174]: f-string

File: pyleecan/Methods/Slot/HoleM63/__init__.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+
+class S63_NoneError(SlotCheckError):
+    """Raised when a propery of HoleM63 is None"""
+
+    pass
+
+
+class S63_WCheckError(SlotCheckError):
+    """ """
```
Comment (AdrienLeduqueEomys) [inline pyleecan/Methods/Slot/HoleM63/__init__.py:11]: docstring

File: pyleecan/Methods/Slot/HoleM63/check.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    -------
+    None
+
+    Raises
+    _______
+    S63_WCheckError
+    """
+
+    # Check that everything is set
+    if self.W0 is None:
```
Comment (AdrienLeduqueEomys) [inline pyleecan/Methods/Slot/HoleM63/check.py:24]: Idem. What happen if W0 and H0 are not set ?

**Comment by BonneelP** [comment]
Thank you for your contribution !

**Pull request merged**